### PR TITLE
Add GenerateShardRanges to vtctl commands

### LINKED ISSUE #6751: Expose a shard range calculator as a vtctl command
#### Feature Description

vtctl should expose a new command, which, given a number of total shards in a keyspace, outputs a list of the hex shard ranges for a keyspace of that size.

Example usage:

```
❯ vtctlclient ListShardRanges -num_shards 2
[
  "-80",
  "80-"
]
```

#### Use Case(s)

From the vitess slack I've seen at least one other person maintaining a tool like this, and I would imagine that other users of vitess have their own solution to this problem. Providing this in vitess removes an additional tool for all those users to maintain, and guarantees that we're all using the same logic.

### PR DESCRIPTION
**Pull request by ajm188**
Resolves #6751

### CODE REVIEW

File: go/vt/vtctl/vtctl.go [hunk 1]
```diff
@@ -378,6 +378,9 @@ var commands = []commandGroup{
 			{"ListTablets", commandListTablets,
 				"<tablet alias> ...",
 				"Lists specified tablets in an awk-friendly way."},
+			{"ListShardRanges", commandListShardRanges,
```
Comment (rafael) [inline go/vt/vtctl/vtctl.go:381]: nit - Maybe "GenerateShardRanges" ? I think List reads as something that already exists.

**Review by rafael — COMMENTED** [review]
From my perspective this is a great idea to have a first class citizen. I've seen this come up multiple times in Slack where people it's not super sure about how to generate the shard ranges for a shard. 

After thinking a bit more about this, we could also add this script to the contrib one. Personally, I think it makes more sense here. 

@deepthi  what do you think about this?

**Comment by deepthi** [comment]
> From my perspective this is a great idea to have a first class citizen. I've seen this come up multiple times in Slack where people it's not super sure about how to generate the shard ranges for a shard.
> 
> After thinking a bit more about this, we could also add this script to the contrib one. Personally, I think it makes more sense here.
> 
> @deepthi what do you think about this?

I think it's a great idea :) +1 on keeping it in the main repo and not in contrib.

File: go/vt/vtctl/vtctl_test.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+			false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := listShardRanges(tt.args.shards)
+			if (err != nil) != tt.wantErr {
```
Comment (deepthi) [inline go/vt/vtctl/vtctl_test.go:61]: May I suggest replacing the more verbose comparisons with `assert` and `require`?

Comment (ajm188) [reply go/vt/vtctl/vtctl_test.go:61]: Absolutely!

File: go/vt/vtctl/vtctl_test.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+			assert.Equal(t, got, tt.want)
+		})
+	}
+}
+
+func TestShardCalculatorForShardsGreaterThan512(t *testing.T) {
+	got, err := generateShardRanges(512)
+	if err != nil {
+		t.Errorf("listShardRanges() error = %v", err)
```
Comment (deepthi) [inline go/vt/vtctl/vtctl_test.go:76]: missed a few

Comment (ajm188) [reply go/vt/vtctl/vtctl_test.go]: 🙈 oops! fixed

File: go/vt/vtctl/vtctl.go [hunk 4]
```diff
@@ -2987,7 +2990,91 @@ func commandWorkflow(ctx context.Context, wr *wrangler.Wrangler, subFlags *flag.
...
+			endKid = fmt.Sprintf(format, end)
+		}
 
+		return fmt.Sprintf("%s-%s", startKid, endKid)
+	}
+
+	start := 0
+	end := 0
+
+	// If shards does not divide evenly into maxShards, then there is some lossiness,
```
Comment (deepthi) [inline go/vt/vtctl/vtctl.go:3047]: Nice implementation and documentation :+1:

**Review by deepthi — APPROVED** [review]
LGTM

**Pull request merged**